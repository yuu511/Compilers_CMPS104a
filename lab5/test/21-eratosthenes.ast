ROOT "<<ROOT>>" 0.0.0 {0} 
FUNCTION "" 6.22.1 {0} void function 
TYPE_ID "" 6.22.1 {0} 
VOID "void" 6.22.1 {0} 
IDENT "fail" 6.22.6 {0} 
PARAM "(" 6.22.11 {0} 
TYPE_ID "" 6.22.12 {1} string variable param lval 
STRING "string" 6.22.12 {0} 
IDENT "expr" 6.22.19 {0} 
TYPE_ID "" 6.22.25 {1} string variable param lval 
STRING "string" 6.22.25 {0} 
IDENT "file" 6.22.32 {0} 
TYPE_ID "" 6.22.38 {1} int variable param lval 
INT "int" 6.22.38 {0} 
IDENT "line" 6.22.42 {0} 
FUNCTION "" 6.24.1 {1} void function 
TYPE_ID "" 6.24.1 {0} 
VOID "void" 6.24.1 {0} 
IDENT "putchr" 6.24.6 {0} 
PARAM "(" 6.24.13 {0} 
TYPE_ID "" 6.24.14 {2} int variable param lval 
INT "int" 6.24.14 {0} 
IDENT "chr" 6.24.18 {0} 
FUNCTION "" 6.25.1 {2} void function 
TYPE_ID "" 6.25.1 {0} 
VOID "void" 6.25.1 {0} 
IDENT "putint" 6.25.6 {0} 
PARAM "(" 6.25.13 {0} 
TYPE_ID "" 6.25.14 {3} int variable param lval 
INT "int" 6.25.14 {0} 
IDENT "num" 6.25.18 {0} 
FUNCTION "" 6.26.1 {3} void function 
TYPE_ID "" 6.26.1 {0} 
VOID "void" 6.26.1 {0} 
IDENT "putstr" 6.26.6 {0} 
PARAM "(" 6.26.13 {0} 
TYPE_ID "" 6.26.14 {4} string variable param lval 
STRING "string" 6.26.14 {0} 
IDENT "str" 6.26.21 {0} 
FUNCTION "" 6.28.1 {4} int function 
TYPE_ID "" 6.28.1 {0} 
INT "int" 6.28.1 {0} 
IDENT "getchr" 6.28.5 {0} 
PARAM "(" 6.28.11 {0} 
FUNCTION "" 6.29.1 {5} string function 
TYPE_ID "" 6.29.1 {0} 
STRING "string" 6.29.1 {0} 
IDENT "getstr" 6.29.8 {0} 
PARAM "(" 6.29.14 {0} 
FUNCTION "" 6.30.1 {6} string function 
TYPE_ID "" 6.30.1 {0} 
STRING "string" 6.30.1 {0} 
IDENT "getln" 6.30.8 {0} 
PARAM "(" 6.30.13 {0} 
FUNCTION "" 7.8.1 {7} int function 
TYPE_ID "" 7.8.1 {0} 
INT "int" 7.8.1 {0} 
IDENT "main" 7.8.5 {0} 
PARAM "(" 7.8.9 {0} 
BLOCK "{" 7.8.12 {8} 
TYPE_ID "" 7.9.4 {8} int variable local lval vreg (7.9.4)
INT "int" 7.9.4 {0} 
IDENT "prime" 7.9.8 {0} 
INTCON "2" 7.9.16 {8} int const 
TYPE_ID "" 7.10.4 {8} int variable local lval vreg (7.10.4)
INT "int" 7.10.4 {0} 
IDENT "index" 7.10.8 {0} 
INTCON "2" 7.10.16 {8} int const 
TYPE_ID "" 7.11.4 {8} int array variable local lval vreg (7.11.4)
ARRAY "array" 7.11.4 {0} 
INT "int" 7.11.10 {0} 
IDENT "sieve" 7.11.15 {0} 
ALLOC "alloc" 7.11.23 {8} int array 
ARRAY "array" 7.11.29 {0} 
INT "int" 7.11.35 {0} 
INTCON "100" 7.11.41 {8} int const 
WHILE "while" 7.13.4 {8} 
LT "<" 7.13.17 {8} int vreg 
IDENT "index" 7.13.11 {8} int variable local lval (7.10.4)
INTCON "100" 7.13.19 {8} int const 
BLOCK "{" 7.13.24 {8} 
'=' "=" 7.14.20 {8} int lval vreg vaddr 
INDEX "[" 7.14.12 {8} int lval vaddr 
IDENT "sieve" 7.14.7 {8} int array variable local lval (7.11.4)
IDENT "index" 7.14.13 {8} int variable local lval (7.10.4)
INTCON "1" 7.14.22 {8} int const 
'=' "=" 7.15.13 {8} int variable local lval vreg 
IDENT "index" 7.15.7 {8} int variable local lval (7.10.4)
'+' "+" 7.15.21 {8} int vreg 
IDENT "index" 7.15.15 {8} int variable local lval (7.10.4)
INTCON "1" 7.15.23 {8} int const 
WHILE "while" 7.18.4 {8} 
LT "<" 7.18.17 {8} int vreg 
IDENT "prime" 7.18.11 {8} int variable local lval (7.9.4)
INTCON "100" 7.18.19 {8} int const 
BLOCK "{" 7.18.24 {8} 
IF "if" 7.19.7 {8} 
INDEX "[" 7.19.16 {8} int lval vaddr 
IDENT "sieve" 7.19.11 {8} int array variable local lval (7.11.4)
IDENT "prime" 7.19.17 {8} int variable local lval (7.9.4)
BLOCK "{" 7.19.25 {8} 
'=' "=" 7.20.16 {8} int variable local lval vreg 
IDENT "index" 7.20.10 {8} int variable local lval (7.10.4)
'*' "*" 7.20.24 {8} int vreg 
IDENT "prime" 7.20.18 {8} int variable local lval (7.9.4)
INTCON "2" 7.20.26 {8} int const 
WHILE "while" 7.21.10 {8} 
LT "<" 7.21.23 {8} int vreg 
IDENT "index" 7.21.17 {8} int variable local lval (7.10.4)
INTCON "100" 7.21.25 {8} int const 
BLOCK "{" 7.21.30 {8} 
'=' "=" 7.22.26 {8} int lval vreg vaddr 
INDEX "[" 7.22.18 {8} int lval vaddr 
IDENT "sieve" 7.22.13 {8} int array variable local lval (7.11.4)
IDENT "index" 7.22.19 {8} int variable local lval (7.10.4)
INTCON "0" 7.22.28 {8} int const 
'=' "=" 7.23.19 {8} int variable local lval vreg 
IDENT "index" 7.23.13 {8} int variable local lval (7.10.4)
'+' "+" 7.23.27 {8} int vreg 
IDENT "index" 7.23.21 {8} int variable local lval (7.10.4)
IDENT "prime" 7.23.29 {8} int variable local lval (7.9.4)
'=' "=" 7.26.13 {8} int variable local lval vreg 
IDENT "prime" 7.26.7 {8} int variable local lval (7.9.4)
'+' "+" 7.26.21 {8} int vreg 
IDENT "prime" 7.26.15 {8} int variable local lval (7.9.4)
INTCON "1" 7.26.23 {8} int const 
'=' "=" 7.29.10 {8} int variable local lval vreg 
IDENT "index" 7.29.4 {8} int variable local lval (7.10.4)
INTCON "2" 7.29.12 {8} int const 
WHILE "while" 7.30.4 {8} 
LT "<" 7.30.17 {8} int vreg 
IDENT "index" 7.30.11 {8} int variable local lval (7.10.4)
INTCON "100" 7.30.19 {8} int const 
BLOCK "{" 7.30.24 {8} 
IF "if" 7.31.7 {8} 
INDEX "[" 7.31.16 {8} int lval vaddr 
IDENT "sieve" 7.31.11 {8} int array variable local lval (7.11.4)
IDENT "index" 7.31.17 {8} int variable local lval (7.10.4)
BLOCK "{" 7.31.25 {8} 
CALL "(" 7.32.17 {2} void function (6.25.1)
IDENT "putint" 7.32.10 {0} 
IDENT "index" 7.32.18 {8} int variable local lval (7.10.4)
CALL "(" 7.33.17 {1} void function (6.24.1)
IDENT "putchr" 7.33.10 {0} 
CHARCON "'\n'" 7.33.18 {8} int const 
'=' "=" 7.35.13 {8} int variable local lval vreg 
IDENT "index" 7.35.7 {8} int variable local lval (7.10.4)
'+' "+" 7.35.21 {8} int vreg 
IDENT "index" 7.35.15 {8} int variable local lval (7.10.4)
INTCON "1" 7.35.23 {8} int const 
RETURN "return" 7.38.4 {8} 
INTCON "0" 7.38.11 {8} int const 
