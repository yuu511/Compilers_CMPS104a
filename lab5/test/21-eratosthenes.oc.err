invalid condition for loop: 7.19.16
invalid condition for loop3ac: invalid statement passed :if
invalid condition for loop: 7.31.16
invalid condition for loop3ac: invalid statement passed :if
NUMBER OF ERRORS REPORTED: 4
