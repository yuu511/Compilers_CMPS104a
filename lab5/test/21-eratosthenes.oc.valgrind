==24231== Memcheck, a memory error detector
==24231== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==24231== Using Valgrind-3.14.0 and LibVEX; rerun with -h for copyright info
==24231== Command: ../oc 21-eratosthenes.oc
==24231== 
==24231== 
==24231== HEAP SUMMARY:
==24231==     in use at exit: 0 bytes in 0 blocks
==24231==   total heap usage: 806 allocs, 806 frees, 166,369 bytes allocated
==24231== 
==24231== All heap blocks were freed -- no leaks are possible
==24231== 
==24231== For counts of detected and suppressed errors, rerun with: -v
==24231== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
