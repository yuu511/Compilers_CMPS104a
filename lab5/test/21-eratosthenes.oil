main:      .function int 
           .local int prime
           .local int index
           .local ptr sieve
           prime = 2  
           index = 2  
           $t0:p = 100 * sizeof int
           sieve = call malloc($t0:p)  
.wh0:      $t1:i = index < 100
           goto .od0 if not $t1:i
.do0:      sieve[index * :i] = 1  
           $t2:i = index + 1
           index = $t2:i  
           goto .wh0
.od0:
.wh1:      $t3:i = prime < 100
           goto .od1 if not $t3:i
           $t4:i = prime + 1
           prime = $t4:i  
           goto .wh1
.od1:
           index = 2  
.wh2:      $t5:i = index < 100
           goto .od2 if not $t5:i
           $t6:i = index + 1
           index = $t6:i  
           goto .wh2
.od2:
           return 0
           return
           .end
