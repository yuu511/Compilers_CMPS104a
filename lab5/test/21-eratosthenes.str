string_set[   0]:    7296690267450279874 0x55b7ba2bad58->"chr"
                   13348923631108115385 0x55b7ba2bb718->"putstr"
string_set[   5]:   12679825168387489523 0x55b7ba2bb278->"num"
string_set[   6]:    9317282556992397945 0x55b7ba2be308->"if"
string_set[   7]:    2297668033614959926 0x55b7ba2befc8->"0"
                    6814028903552918164 0x55b7ba2bcd88->"array"
string_set[   9]:   17581720658427701949 0x55b7ba2bcf48->"sieve"
string_set[  10]:    6637313742931709005 0x55b7ba2be868->"*"
                    1583576014053361350 0x55b7ba2bb8d8->"str"
                   13497578857207247412 0x55b7ba2ba318->"string"
                   11597697714117577063 0x55b7ba2bc8d8->"="
string_set[  11]:    1396094175042499165 0x55b7ba2bcdc8->"<"
string_set[  13]:   12868445110721718657 0x55b7ba2ba978->")"
string_set[  16]:   10159970873491820195 0x55b7ba2bdc08->"1"
                   11911796782241914524 0x55b7ba2b9088->"void"
                    4698864730501174006 0x55b7ba2bcaf8->"index"
string_set[  18]:   13581035385482130682 0x55b7ba2bd788->"["
string_set[  19]:    5344662657899890615 0x55b7ba2baa38->";"
string_set[  20]:    1894629132964850000 0x55b7ba2bc818->"prime"
                   15816506932648033945 0x55b7ba2ba258->"("
                   11868120863286193964 0x55b7ba2bc518->"main"
string_set[  23]:    6340238197210063670 0x55b7ba2bd448->"100"
string_set[  25]:   11166701219533395393 0x55b7ba2bbef8->"getstr"
string_set[  29]:   14635384904416252381 0x55b7ba2c02b8->"return"
                    1370730858159036685 0x55b7ba2bde68->"+"
string_set[  34]:    7211130878751977441 0x55b7ba2bab98->"putchr"
                    8290819294854170237 0x55b7ba2bd008->"alloc"
string_set[  37]:    6192799629375248351 0x55b7ba2bff98->"'\n'"
                   12291392672316406109 0x55b7ba2ba898->"line"
string_set[  39]:   18413295556186398326 0x55b7ba2bb0b8->"putint"
string_set[  40]:   10319632053171904911 0x55b7ba2bd5c8->"while"
string_set[  44]:      21606623943142546 0x55b7ba2bc6d8->"{"
string_set[  45]:    6142509188972423790 0x55b7ba2ba498->""
string_set[  47]:   13686721134187162920 0x55b7ba2be008->"}"
string_set[  53]:   12051031859636400188 0x55b7ba2bd8c8->"]"
string_set[  57]:    9386170080689500517 0x55b7ba2ba3d8->"expr"
string_set[  58]:    4017852018407011111 0x55b7ba2ba658->"file"
                    3729804957429652673 0x55b7ba2b3d08->"<<ROOT>>"
string_set[  59]:   16579916199511856337 0x55b7ba2bc1f8->"getln"
string_set[  61]:    4551451650890805270 0x55b7ba2bc998->"2"
string_set[  63]:    9611428142580219387 0x55b7ba2ba158->"fail"
string_set[  64]:   12597418120160561823 0x55b7ba2ba7d8->"int"
string_set[  70]:    9201800968376702500 0x55b7ba2bbbf8->"getchr"
string_set[  71]:   14526079891305879776 0x55b7ba2bcf08->">"
string_set[  78]:    4706206694200543916 0x55b7ba2ba198->","
load_factor = 0.570
bucket_count = 79
max_bucket_size = 4
