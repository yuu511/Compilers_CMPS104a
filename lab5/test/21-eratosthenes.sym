getln (6.30.1) {6} string function 

main (7.8.1) {7} int function 
   prime (7.9.4) {8} int variable local lval 0
   index (7.10.4) {8} int variable local lval 1
   sieve (7.11.4) {8} int array variable local lval 2

putchr (6.24.1) {1} void function 

fail (6.22.1) {0} void function 

putint (6.25.1) {2} void function 

putstr (6.26.1) {3} void function 

getstr (6.29.1) {5} string function 

getchr (6.28.1) {4} int function 

