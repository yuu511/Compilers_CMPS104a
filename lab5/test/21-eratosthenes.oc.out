CONDITIONAL WHILE STATEMENT SYMBOL TOK_LT 
CONDITIONAL WHILE STATEMENT SYMBOL TOK_LT 
CONDITIONAL IF STATEMENT SYMBOL TOK_INDEX 
CONDITIONAL WHILE STATEMENT SYMBOL TOK_LT 
CONDITIONAL WHILE STATEMENT SYMBOL TOK_LT 
CONDITIONAL IF STATEMENT SYMBOL TOK_INDEX 
